extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/BooleanChunkInputStreamGenerator.java

```diff
@@ -0,0 +1,280 @@
+                        ++cachedNullCount;
+                    }
+                });
+            }
+            return cachedNullCount;
+        }
+
+        @Override
+        protected int getRawSize() {
+            long size = 0;
+            if (sendValidityBuffer()) {
+                size += getValidityMapSerializationSizeFor(subset.intSize(DEBUG_NAME));
+            }
+            size += getNumLongsForBitPackOfSize(subset.intSize(DEBUG_NAME)) * 8L;
+            return LongSizedDataStructure.intSize("BaseChunkInputStream.getRawSize", size);
+        }
+
+        @Override
+        public void visitFieldNodes(final FieldNodeListener listener) {
+            listener.noteLogicalFieldNode(subset.intSize(DEBUG_NAME), nullCount());
+        }
+
+        @Override
+        public void visitBuffers(final BufferListener listener) {
+            // validity
+            int validityLen = sendValidityBuffer() ? getValidityMapSerializationSizeFor(subset.intSize(DEBUG_NAME)) : 0;
+            listener.noteLogicalBuffer(validityLen);
+            // payload
+            listener.noteLogicalBuffer(getNumLongsForBitPackOfSize(subset.intSize(DEBUG_NAME)) * 8L);
+        }
+
+        @Override
+        @SuppressWarnings("UnstableApiUsage")
+        public int drainTo(final OutputStream outputStream) throws IOException {
+            if (read || subset.isEmpty()) {
+                return 0;
+            }
+
+            long bytesWritten = 0;
+            read = true;
+            final LittleEndianDataOutputStream dos = new LittleEndianDataOutputStream(outputStream);
+            // write the validity array with LSB indexing
+            final SerContext context = new SerContext();
+            final Runnable flush = () -> {
+                try {
+                    dos.writeLong(context.accumulator);
+                } catch (final IOException e) {
+                    throw new UncheckedDeephavenException("Unexpected exception while draining data to OutputStream: ", e);
+                }
+                context.accumulator = 0;
+                context.count = 0;
+            };
+
+            if (sendValidityBuffer()) {
+                subset.forAllRowKeys(row -> {
+                    if (chunk.get((int) row) != NULL_BYTE) {
+                        context.accumulator |= 1L << context.count;
+                    }
+                    if (++context.count == 64) {
+                        flush.run();
+                    }
+                });
+                if (context.count > 0) {
+                    flush.run();
+                }
+                bytesWritten += getValidityMapSerializationSizeFor(subset.intSize(DEBUG_NAME));
+            }
+
+            // write the included values
+            subset.forAllRowKeys(row -> {
+                final byte byteValue = chunk.get((int) row);
+                if (byteValue != NULL_BYTE) {
+                    context.accumulator |= (byteValue & 0x1L) << context.count;
```

Comment:
Is `byteValue`always in [0,1]?  If `byteValue` allowed to be `>1` this may encode 0 for `true` values.  I'm not familiar with the underlying storage for our boolean data.

Reply:
This is an excellent catch. I did not look at both conversion methods. You are right, byte values greater than 0 are all true.